infection_monkey/windows_upgrader.py

```diff
@@ -0,0 +1,67 @@
+import logging
+import os
+import struct
+import subprocess
+import sys
+
+import time
+
+import monkeyfs
+from config import WormConfiguration
+from control import ControlClient
+from exploit.tools import build_monkey_commandline_explicitly
+from model import MONKEY_CMDLINE_WINDOWS
+
+__author__ = 'itay.mizeretz'
+
+LOG = logging.getLogger(__name__)
+
+if "win32" == sys.platform:
+    from win32process import DETACHED_PROCESS
+else:
+    DETACHED_PROCESS = 0
+
+
+class WindowsUpgrader(object):
```

Comment:
Many of these functions should be moved to a generic utils file, I can think of other reasons to have them.

Reply:
Fixed: Moved to utils.